trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1255,7 +1326,84 @@ public Response deleteDerivedOutput(
+        }
+
+        if (derivedOutputId == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_OUTPUTID + " (derived)").build(); //$NON-NLS-1$
+        }
+
+        try (FlowScopeLog scope = new FlowScopeLogBuilder(LOGGER, Level.FINE, "DataProviderService#removeDataProvider") //$NON-NLS-1$
+                .setCategory(outputId).build()) {
+            TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+            if (experiment == null) {
+                return Response.status(Status.NOT_FOUND).entity(NO_SUCH_TRACE).build();
+            }
+
+            IDataProviderDescriptor parentDescriptor = getDescriptor(experiment, outputId);
+            if (parentDescriptor == null) {
+                return Response.status(Status.NOT_FOUND).entity(NO_SUCH_PROVIDER + " (source): " + outputId).build(); //$NON-NLS-1$
+            }
+
+            IDataProviderDescriptor derivedDescriptor = getDescriptor(experiment, derivedOutputId);
+            if (derivedDescriptor == null) {
+                return Response.status(Status.NOT_FOUND).entity(NO_SUCH_PROVIDER + " (derived): " + derivedOutputId).build(); //$NON-NLS-1$
+            }
+
+            IDataProviderFactory factory = manager.getFactory(outputId);
+            if (factory == null) {
+                return Response.status(Status.METHOD_NOT_ALLOWED).entity(NO_SUCH_PROVIDER).build();
+            }
+
+            ITmfDataProviderConfigurator configurator = factory.getAdapter(ITmfDataProviderConfigurator.class);
+            if (configurator == null) {
+                return Response.status(Status.METHOD_NOT_ALLOWED).entity(NO_SUCH_PROVIDER).build();
+            }
+            configurator.removeDataProviderDescriptor(experiment, derivedDescriptor);
+            return Response.ok().build();
+        } catch (TmfConfigurationException e) {
+            return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+        }
+    }
+
+    private static Response validateParameters(String outputId, QueryParameters queryParameters) {
+        if (outputId == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_OUTPUTID).build();
+        }
+        if (queryParameters == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_PARAMETERS).build();
+        }
+        return null;
+    }
+
+    private static Response validateOutputConfigParameters(String outputId, ConfigurationQueryParameters queryParameters) {
+        if (outputId == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_OUTPUTID).build();
+        }
+        if (queryParameters == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_PARAMETERS).build();
+        }
+
+        Map<String, Object> parameters = queryParameters.getParameters();
+        if (!(parameters.get(DataProviderParameterUtils.CONFIGURATION_TYPE_ID) instanceof String)){
```

Comment:
This should be queryParameters.getTypeId() ?
edit: But this is only introduced in the next commit.
edit 2: Then you can probably remove the "typeId" constant in DataProviderParameterUtils?